File: solutions/007/007.cc
#include <iostream>

int main() {
    int n = 10001, lenPrimes = 1;
    int primes[n];
    primes[0] = 2;

    for (int x = 3; lenPrimes < n+1; x += 2) {
        for (int i = 0; i < lenPrimes; i++)
            if (x % primes[i] == 0)
                break;
            else if (primes[i]*primes[i] > x) {
                primes[lenPrimes++] = x;
                break;
            }
    }
    std::cout << primes[n-1] << std::endl;
}

File: solutions/007/_007.cs
using System;

public class _007 {
    public static void Main(string[] args) {
        int[] primes = new int[10001];
        primes[0] = 2;
        int lenPrimes = 1;
        for (int x = 3; lenPrimes < 10001; x += 2)
            for (int i = 0; i < lenPrimes; i++)
                if (x % primes[i] == 0)
                    break;
                else if (primes[i] * primes[i] > x) {
                    primes[lenPrimes++] = x;
                    break;
                }
        Console.WriteLine(primes[primes.Length - 1]);
    }
}
